pkg/models/devops/devops.go

```diff
@@ -310,25 +296,124 @@ func GetCrumb(req *http.Request) (*Crumb, error) {
 	return res, err
 }
 
-// jenkins request and parse response
-func jenkinsClient(baseUrl string, req *http.Request, res interface{}) error {
-	resBody, err := Client(baseUrl, req)
+func CheckScriptCompile(req *http.Request) ([]byte, error) {
+	baseUrl := JenkinsUrl + CheckScriptCompileUrl
+	log.Infof("Jenkins-url: " + baseUrl)
+
+	resBody, err := sendJenkinsRequest(baseUrl, req)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+
+	return resBody, err
+}
+
+func CheckCron(req *http.Request) (*CheckCronRes, error) {
+	baseUrl := JenkinsUrl + CheckCronUrl + req.URL.RawQuery
+	log.Infof("Jenkins-url: " + baseUrl)
+	var res = new(CheckCronRes)
+
+	resp, err := http.Get(baseUrl)
```

Comment:
add basic auth,  only admin could access init-job

Reply:
check cron and check pipeline script only admin?

Reply:
use admin  account in apiserver, but all user could use api

Reply:
ok